CourseSearchBuilderBlobTrigger/__init__.py

```diff
@@ -0,0 +1,65 @@
+import os
+import logging
+import requests
+import traceback
+
+from datetime import datetime
+
+import azure.functions as func
+from SharedCode import utils, exceptions
+from . import search, models, helpers
+
+def main(dataset: func.InputStream):
+    logging.info(f"Python blob trigger function processed blob \n"
+                 f"Name: {dataset.name}\n"
+                 f"Blob Size: {dataset.length} bytes\n")
+
+    api_key = os.environ['SearchAPIKey']
+    search_url = os.environ['SearchURL']
+    api_version = os.environ['AzureSearchAPIVersion']
+
+    try:
+        headers = {'Content-Type': 'application/json',
+            'api-key': api_key,
+            'odata': 'verbose' }
+
+        query_string = '?api-version=' + api_version
+
+        # Get version number
+        version = helpers.get_version(dataset.name)
+
+        index_name = f"courses-{version}"
+    
+        # Delete search index if it already exists
+        delete_url = search_url + "/indexes/" + index_name + query_string
+
+        search.delete_index(delete_url, headers, index_name)
+
+        # Create search index
+        index_schema = search.get_index_schema(index_name)
+        url = search_url + "/indexes" + query_string
+        search.create_index(url, headers, index_schema, index_name)

```

Comment:
Maybe simpler if class - as here pulling the schema from the search module and then passing it back?